### PR TITLE
Make stardoc our documentation tool

### LINKED ISSUE #337: Documentation needs to be updated with new features
The documentation is out of date due to recent PRs. Can this be updated?

### PR DESCRIPTION
**Pull request by damienmg**
The current documentation was difficult to maintain and not in sync with the inline one.

These changes adjust the setting so we can generate the documentation in markdown format and it will continue to be served on Github pages, this time with an easy way to sync it.

Each commit of this PR have been made to be reviewed separately.

Fixes #337

### CODE REVIEW

File: .bazelci/presubmit.yml [hunk 1]
```diff
@@ -50,6 +47,12 @@ tasks:
     working_directory: examples
     test_targets:
     - //...
+  docs:
+    name: Docs
+    platform: ubuntu1804
```
Comment (mfarrugi) [inline .bazelci/presubmit.yml:52]: Does this actually care about the ubuntu version and/or not work on osx?

File: README.md [hunk 2]
```diff
@@ -11,97 +11,4 @@ This repository provides rules for building [Rust][rust] projects with [Bazel](h
...
-
-Currently the most common approach to managing external dependencies is using
-[cargo-raze](https://github.com/google/cargo-raze) to generate `BUILD` files for Cargo crates.
-
-<a name="roadmap"></a>
-## Roadmap
-
-* Improve expressiveness of features and support for [Cargo's feature groups](http://doc.crates.io/manifest.html#the-[features]-section).
-* Add `cargo_crate` workspace rule for pulling crates from [Cargo](https://crates.io/).
+Please refer to [the full documentation](https://bazelbuild.github.io/rules_rust).
```
Comment (mfarrugi) [inline README.md:14]: Github pages version looks like it was never updated, how is that done?

Might be easier to link into the repo if that's not trivial.

File: docs/WORKSPACE [hunk 3]
```diff
@@ -1 +1,38 @@
...
+load("@io_bazel_rules_rust//proto:repositories.bzl", "rust_proto_repositories")
+load("@io_bazel_rules_rust//:workspace.bzl", "bazel_version")
+load("@io_bazel_stardoc//:setup.bzl", "stardoc_repositories")
+
+rust_repositories()
+rust_proto_repositories()
+bazel_version(name = "bazel_version")
+stardoc_repositories()
+
+local_repository(
```
Comment (mfarrugi) [inline docs/WORKSPACE:35]: This is because upstream is missing `bzl_library`?

File: README.md (hunk 2 above)
Comment (damienmg) [reply README.md:14]: This is just done by modifying this directory, no needs for other change normally.

File: .bazelci/presubmit.yml (hunk 1 above)
Comment (damienmg) [reply .bazelci/presubmit.yml:52]: It works on OSX I have developed it on OSX. Added the test for OS X just to make sure we don't break our own development workflow.

File: docs/WORKSPACE (hunk 3 above)
Comment (damienmg) [reply docs/WORKSPACE:35]: Correct, I don't see it has being possible to upstream as rules_proto has 0 deps (in addition to being shipped along with bazel). Added a comment.

**Pull request merged**